WordPress/Classes/ViewRelated/Post/WPPostViewController.m

```diff
@@ -209,6 +210,8 @@ - (instancetype)initWithPost:(AbstractPost *)post
     BOOL changeToEditModeDueToUnsavedChanges = (mode == kWPEditorViewControllerModePreview
                                                 && [post hasUnsavedChanges]);
     
+    self.isOpenedDirectlyForEditing = (mode == kWPEditorViewControllerModeEdit);
```

Comment:
Setting an ivar outside of the `if (self)` was clearly incorrect. Sorry about that. I understand why you'd like to not reference `self` in `init` as well. I've edited the commits to incorporate these changes.